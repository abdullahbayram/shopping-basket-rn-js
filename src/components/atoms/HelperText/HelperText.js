import * as React from 'react';
import PropTypes from 'prop-types';
import { HelperText as PHelperText } from 'react-native-paper';
import { StyleSheet } from 'react-native';

const HelperText = ({ children, type, visible, style }) => (
  <PHelperText type={type} visible={visible} style={[styles.helperText, style]}>
    {children}
  </PHelperText>
);

HelperText.propTypes = {
  children: PropTypes.node.isRequired,
  type: PropTypes.string,
  visible: PropTypes.bool,
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
};

export default HelperText;

const styles = StyleSheet.create({
  helperText: {
    paddingVertical: 0,
  },
});
