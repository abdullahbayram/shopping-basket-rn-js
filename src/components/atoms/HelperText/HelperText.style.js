import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  helperText: {
    paddingVertical: 0,
  },
});

export default styles;
