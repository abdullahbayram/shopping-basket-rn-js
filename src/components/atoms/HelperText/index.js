import * as React from 'react';
import PropTypes from 'prop-types';
import { HelperText as PHelperText } from 'react-native-paper';
import styles from './HelperText.style';

const HelperText = ({ children, type, visible, style }) => (
  <PHelperText type={type} visible={visible} style={[styles.helperText, style]}>
    {children}
  </PHelperText>
);

HelperText.propTypes = {
  children: PropTypes.node.isRequired,
  type: PropTypes.string,
  visible: PropTypes.bool,
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
};

export default HelperText;
