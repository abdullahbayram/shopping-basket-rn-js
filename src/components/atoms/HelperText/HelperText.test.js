import React from 'react';
import { render, screen } from '@testing-library/react-native';
import HelperText from '.';

jest.mock('./HelperText.style', () => ({
  helperText: {
    color: 'red',
    fontSize: 12,
  },
}));

describe('HelperText Component', () => {
  it('should match the snapshot when visible is true', () => {
    const { toJSON } = render(
      <HelperText type="info" visible>
        Visible helper text
      </HelperText>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match the snapshot when visible is false', () => {
    const { toJSON } = render(
      <HelperText type="info" visible={false}>
        Hidden helper text
      </HelperText>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should render correctly with required props', () => {
    render(
      <HelperText type="error" visible style={{ fontWeight: 'bold' }}>
        This is a helper text
      </HelperText>,
    );

    const helperText = screen.getByText('This is a helper text');

    expect(helperText).toBeTruthy();
    expect(helperText.props.style).toEqual(expect.objectContaining({ fontWeight: 'bold' }));
  });

  it('should render with default and custom styles merged', () => {
    render(
      <HelperText type="info" visible style={{ letterSpacing: 1 }}>
        Default styled text
      </HelperText>,
    );

    const defaultText = screen.getByText('Default styled text');

    expect(defaultText).toBeTruthy();
    expect(defaultText.props.style).toEqual(
      expect.objectContaining({
        color: 'red',
        fontSize: 12,
      }),
      expect.objectContaining({ letterSpacing: 1 }),
    );
  });
});
