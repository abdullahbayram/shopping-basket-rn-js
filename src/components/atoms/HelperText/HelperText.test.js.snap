// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelperText Component should match the snapshot when visible is false 1`] = `
<Text
  collapsable={false}
  maxFontSizeMultiplier={1.5}
  onLayout={[Function]}
  style={
    {
      "color": "red",
      "fontFamily": "System",
      "fontSize": 12,
      "fontWeight": "400",
      "letterSpacing": 0.25,
      "lineHeight": 20,
      "opacity": 0,
      "paddingHorizontal": 12,
      "paddingVertical": 4,
      "textAlign": "left",
      "transform": [],
      "writingDirection": "ltr",
    }
  }
>
  Hidden helper text
</Text>
`;

exports[`HelperText Component should match the snapshot when visible is true 1`] = `
<Text
  collapsable={false}
  maxFontSizeMultiplier={1.5}
  onLayout={[Function]}
  style={
    {
      "color": "red",
      "fontFamily": "System",
      "fontSize": 12,
      "fontWeight": "400",
      "letterSpacing": 0.25,
      "lineHeight": 20,
      "opacity": 1,
      "paddingHorizontal": 12,
      "paddingVertical": 4,
      "textAlign": "left",
      "transform": [],
      "writingDirection": "ltr",
    }
  }
>
  Visible helper text
</Text>
`;
